public/util/lex.ts

```diff
@@ -197,79 +379,120 @@ export function filterParamsToLexQuery(
   );
 
   const variableDict = buildVariableDictionary(allVariables);
-  const filterVariables = [];
+  const filterVariables = new Map<string, Variable[]>();
+  // check that the filter variables exist
   decodedFilters.forEach((f) => {
     if (variableDict[f.key]) {
-      filterVariables.push(variableDict[f.key]);
+      if (filterVariables.has(f.key)) {
+        filterVariables.get(f.key).push(variableDict[f.key]);
+        return;
+      }
+      filterVariables.set(f.key, [variableDict[f.key]]);
     }
   });
-  const highlightVariables = [];
+  const highlightVariables = new Map<string, Variable[]>();
   decodedHighlights.forEach((h) => {
     if (variableDict[h.key]) {
-      highlightVariables.push(variableDict[h.key]);
+      if (highlightVariables.has(h.key)) {
+        highlightVariables.get(h.key).push(variableDict[h.key]);
+        return;
+      }
+      highlightVariables.set(h.key, [variableDict[h.key]]);
     }
   });
 
   let activeVariables = [
-    ...highlightVariables,
-    ...filterVariables,
-  ] as Variable[];
+    ...Array.from(highlightVariables.values()).map((hv) => {
+      return { variable: hv[0], count: hv.length, mode: INCLUDE_FILTER };
+    }),
+    ...Array.from(filterVariables.values()).map((fv) => {
+      return { variable: fv[0], count: fv.length, mode: EXCLUDE_FILTER };
+    }),
+  ] as VariableInfo[];
   // remove timeseries
   activeVariables = activeVariables.filter((v) => {
-    return v.colType !== TIMESERIES_TYPE;
+    return v.variable.colType !== TIMESERIES_TYPE;
   });
   const activeVariablesMap = new Map(
     activeVariables.map((v) => {
-      return [v.key, true];
+      return [v.variable.key, true];
     })
   );
+  const highlightMap = new Map<string, Filter[]>();
+  const filterMap = new Map<string, Filter[]>();
+  decodedHighlights.forEach((el) => {
+    if (activeVariablesMap.has(el.key)) {
+      if (highlightMap.has(el.key)) {
+        highlightMap.get(el.key).push(el);
+        return;
+      }
+      highlightMap.set(el.key, [el]);
+    }
+  });
+  decodedFilters.forEach((el) => {
+    if (activeVariablesMap.has(el.key)) {
+      if (filterMap.has(el.key)) {
+        filterMap.get(el.key).push(el);
+        return;
+      }
+      filterMap.set(el.key, [el]);
+    }
+  });
+  return { activeVariables, highlightMap, filterMap };
+}
+export function filterParamsToLexQuery(templateInfo: TemplateInfo) {
```

Comment:
return type missing?

Reply:
This function directly interfaces with lex.js and is a builder for the query. The return type is dynamic and as far as I am aware it is not possible to type. The main issue is the properties scale. For example, if there are 3 highlights with the same variable type there will be 3 max and min properties (min_0, min_1, min_2, max_0, max_1, max_2).